src/esp/assets/ResourceManager.cpp

```diff
@@ -1147,23 +1149,42 @@ void ResourceManager::loadMaterials(Importer& importer,
     // as long as the materialName includes the full path to the material
     Cr::Containers::Optional<Mn::Trade::MaterialData> materialData =
         importer.material(iMaterial);
-    if (!materialData ||
-        !(materialData->types() & Magnum::Trade::MaterialType::Phong)) {
+
+    if (!materialData) {
       LOG(ERROR) << "Cannot load material, skipping";
       continue;
     }
 
-    const auto& phongMaterialData =
-        static_cast<Mn::Trade::PhongMaterialData&>(*materialData);
     std::unique_ptr<gfx::MaterialData> finalMaterial;
     int textureBaseIndex = loadedAssetData.meshMetaData.textureIndex.first;
-    if (loadedAssetData.assetInfo.requiresLighting) {
-      finalMaterial =
-          buildPhongShadedMaterialData(phongMaterialData, textureBaseIndex);
 
+    if (loadedAssetData.assetInfo.requiresLighting &&
+        materialData->types() &
+            Magnum::Trade::MaterialType::PbrMetallicRoughness) {
+      const Mn::Trade::PbrMetallicRoughnessMaterialData&
+          pbrMetallicRoughnessMaterialData =
+              static_cast<Mn::Trade::PbrMetallicRoughnessMaterialData&>(
+                  *materialData);
+
+      finalMaterial = gfx::buildPhongFromPbrMetallicRoughness(
+          pbrMetallicRoughnessMaterialData, textureBaseIndex, textures_);
     } else {
-      finalMaterial =
-          buildFlatShadedMaterialData(phongMaterialData, textureBaseIndex);
+      ASSERT(materialData);
+      if (!(materialData->types() & Magnum::Trade::MaterialType::Phong)) {
+        LOG(ERROR) << "Cannot load material, skipping";
+        continue;
+      }
+
+      const auto& phongMaterialData =
+          static_cast<Mn::Trade::PhongMaterialData&>(*materialData);
+      if (loadedAssetData.assetInfo.requiresLighting) {
+        finalMaterial =
+            buildPhongShadedMaterialData(phongMaterialData, textureBaseIndex);
```

Comment:
Just to be clear, this code path using `buildPhongShadedMaterialData` is only if we import a Phong material (not a PBR material).

Reply:
To clarify: right now, for [backwards compatibility purposes](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1TinyGltfImporter.html#Trade-TinyGltfImporter-behavior-materials), all PBR materials will be Phong materials as well.